pkg/volume/util/atomic_writer.go

```diff
@@ -355,8 +359,15 @@ func (w *AtomicWriter) newTimestampDir() (string, error) {
 
 // writePayloadToDir writes the given payload to the given directory.  The
 // directory must exist.
-func (w *AtomicWriter) writePayloadToDir(payload map[string][]byte, dir string) error {
+func (w *AtomicWriter) writePayloadToDir(payload map[string][]byte, mode map[string]int32, dir string) error {
 	for userVisiblePath, content := range payload {
+		perm := w.defaultMode
+		if mode[userVisiblePath] != 0 {
+			perm = os.FileMode(mode[userVisiblePath])
+		}
+		// TODO: move this to validation or some other place
+		perm &= 0777
```

Comment:
This is what I meant when I list the pendings things in a comment on the PR and listed "validation" :)